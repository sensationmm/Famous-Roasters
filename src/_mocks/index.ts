import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
