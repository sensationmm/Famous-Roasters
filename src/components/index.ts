import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
