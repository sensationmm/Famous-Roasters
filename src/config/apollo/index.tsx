import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
