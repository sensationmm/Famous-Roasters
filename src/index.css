@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}