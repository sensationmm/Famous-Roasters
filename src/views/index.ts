import { About } from './About'
import { Account } from './Account'
import { Cart } from './Cart'
import { Catalogue } from './Catalogue'
import { Checkout } from './Checkout'
import { Contact } from './Contact'
import { Error } from './Error'
import { Home } from './Home'
import { OurRoasters } from './OurRoasters'
import { Product } from './Product'
import { TasteFinder } from './TasteFinder'

export { About as About }
export { Account as Account }
export { Cart as Cart }
export { Catalogue as Catalogue }
export { Checkout as Checkout }
export { Contact as Contact }
export { Error as Error }
export { Home as Home }
export { OurRoasters as OurRoasters }
export { Product as Product }
export { TasteFinder as TasteFinder }
