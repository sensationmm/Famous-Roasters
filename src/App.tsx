import { ApolloProvider } from '@apollo/client'
import * as Sentry from '@sentry/react'
import React, { useEffect, useLayoutEffect, useState } from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom'
import { CartProvider, Loader } from 'src/components'
import { famousRoastersClient, hygraphClient, i18n, storeFrontClient } from 'src/config'
import {
  Account,
  Auth,
  Blog,
  BlogListByCategory,
  Cart,
  Catalogue,
  CategoryList,
  Error,
  FeaturedProduct,
  Orders,
  Product,
  Profile,
  TasteFinder,
} from 'src/views'

import LoadingContext from './hooks/isLoading'
import ScrollToTop from './ScrollToTop'

const App = () => {
  const [isBlog, setIsBlog] = useState(false)
  const [isLoading, setIsLoading] = useState(false)

  useLayoutEffect(() => {
    const host = window.location.host
    setIsBlog(host.split('.').includes('blog'))
  }, [])

  useEffect(() => {
    document.body.style.overflow = isLoading ? 'hidden' : 'auto'
  }, [isLoading])

  return (
    <ApolloProvider client={storeFrontClient()}>
      <CartProvider>
        <LoadingContext.Provider value={{ isLoading, setIsLoading }}>
          <I18nextProvider i18n={i18n}>
            <BrowserRouter>
              <ScrollToTop />
              {isLoading && (
                <div className="fixed flex justify-center items-center top-0 left-0 w-screen h-screen bg-brand-grey-whisper/60 z-[1000]">
                  <Loader />
                </div>
              )}
              <Routes>
                <Route
                  path="/blog/en"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <CategoryList locale="de_en" />
                    </ApolloProvider>
                  }
                />
                <Route
                  path="/blog/de"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <CategoryList locale="de_de" />
                    </ApolloProvider>
                  }
                />

                <Route
                  path="/blog/en/:category"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <BlogListByCategory locale="de_en" />
                    </ApolloProvider>
                  }
                />
                <Route
                  path="/blog/de/:category"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <BlogListByCategory locale="de_de" />
                    </ApolloProvider>
                  }
                />

                <Route
                  path="/blog/en/:category/:slug"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <Blog locale="de_en" />
                    </ApolloProvider>
                  }
                />

                <Route
                  path="/blog/de/:category/:slug"
                  element={
                    <ApolloProvider client={hygraphClient()}>
                      <Blog locale="de_de" />
                    </ApolloProvider>
                  }
                />

                <Route path="/blog" element={<Navigate to="/blog/de" replace />} />

                <Route path="/cart" element={<Cart />} />
                <Route path="/catalogue" element={<Catalogue />} />
                <Route path="/catalogue/:productType" element={<Catalogue />} />
                <Route path="/product/:id" element={<Product />} />
                <Route
                  path="/taste-finder"
                  element={
                    <ApolloProvider client={famousRoastersClient()}>
                      <TasteFinder />
                    </ApolloProvider>
                  }
                />
                <Route path="/featured/:id" element={<FeaturedProduct />} />

                <Route path="/login" element={<Auth authState={'signIn'} />} />
                <Route path="/register" element={<Auth authState={'signUp'} />} />
                <Route path="/register-confirm" element={<Auth authState={'confirmSignUp'} />} />
                <Route path="/reset-password" element={<Auth authState={'forgotPassword'} />} />
                <Route
                  path="/profile"
                  element={
                    <ApolloProvider client={famousRoastersClient()}>
                      <Profile />
                    </ApolloProvider>
                  }
                />
                <Route
                  path="/orders"
                  element={
                    <ApolloProvider client={famousRoastersClient()}>
                      <Orders />
                    </ApolloProvider>
                  }
                />
                <Route path="/account" element={<Account />} />
                <Route path="*" element={<Error />} />
              </Routes>
            </BrowserRouter>
          </I18nextProvider>
        </LoadingContext.Provider>
      </CartProvider>
    </ApolloProvider>
  )
}

export default Sentry.withProfiler(App)
