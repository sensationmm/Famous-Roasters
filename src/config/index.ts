import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
