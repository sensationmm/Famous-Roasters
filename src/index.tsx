import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
