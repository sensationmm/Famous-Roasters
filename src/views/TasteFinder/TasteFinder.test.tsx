import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { MemoryRouter } from 'react-router-dom'
import { i18n } from 'src/config'

import { TasteFinder } from '.'

describe('Taste Finder view', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <MemoryRouter initialEntries={['/taste-finder']}>
          <TasteFinder />
        </MemoryRouter>
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can navigate throughout the screens', async () => {
    render(
      <I18nextProvider i18n={i18n}>
        <MemoryRouter initialEntries={['/taste-finder']}>
          <TasteFinder />
        </MemoryRouter>
      </I18nextProvider>,
    )
    const startButton = await screen.findByTestId('button-start')
    expect(startButton).toBeInTheDocument()
    fireEvent.click(startButton)
    const input = await screen.findByTestId('your-name-input')
    expect(input).toBeInTheDocument()
    fireEvent.click(input)
    fireEvent.change(input, { target: { value: 'Flávi' } })
    const nextButton = await screen.findByTestId('nextButton')
    expect(nextButton).toBeInTheDocument()
    fireEvent.click(nextButton)
    const prevButton = await screen.findByTestId('prevButton')
    expect(prevButton).toBeInTheDocument()
    fireEvent.click(prevButton)
    fireEvent.click(input)
    fireEvent.change(input, { target: { value: 'Flávio' } })
    expect(nextButton).toBeInTheDocument()
    fireEvent.click(nextButton)
    const imageCheckboxesBitterness = await screen.findAllByTestId('image-checkbox')
    expect(imageCheckboxesBitterness[0]).toBeInTheDocument()
    fireEvent.click(imageCheckboxesBitterness[0])
    fireEvent.click(nextButton)
    const imageCheckboxesSweetness = await screen.findAllByTestId('image-checkbox')
    expect(imageCheckboxesSweetness[0]).toBeInTheDocument()
    fireEvent.click(imageCheckboxesSweetness[0])
    fireEvent.click(nextButton)
    const imageCheckboxesAcidity = await screen.findAllByTestId('image-checkbox')
    expect(imageCheckboxesAcidity[0]).toBeInTheDocument()
    fireEvent.click(imageCheckboxesAcidity[0])
    fireEvent.click(nextButton)
    const imageCheckboxesBody = await screen.findAllByTestId('image-checkbox')
    expect(imageCheckboxesBody[0]).toBeInTheDocument()
    fireEvent.click(imageCheckboxesBody[0])
    fireEvent.click(nextButton)
    const iconCheckboxBrewing = await screen.findAllByTestId('icon-checkbox')
    expect(iconCheckboxBrewing[0]).toBeInTheDocument()
    fireEvent.click(iconCheckboxBrewing[0])
    fireEvent.click(nextButton)
    const iconCheckboxAdventurous = await screen.findAllByTestId('icon-checkbox')
    expect(iconCheckboxAdventurous[0]).toBeInTheDocument()
    fireEvent.click(iconCheckboxAdventurous[0])
    fireEvent.click(nextButton)
  })
})
