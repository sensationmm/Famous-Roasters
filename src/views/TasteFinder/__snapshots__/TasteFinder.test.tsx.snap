// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taste Finder view Renders correctly 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col text-brand-black"
  >
    <div
      class="bg-white z-10"
    >
      <header
        class="relative bg-brand-black"
      >
        <nav
          aria-label="Top"
          class="max-w-7xl mx-auto px-4 md:px-6 xl:px-8"
        >
          <div
            class="h-16 flex items-center justify-between border-b border-brand-black"
          >
            <div
              class="flex-1 flex items-center xl:hidden"
            >
              <button
                class="-ml-2 bg-brand-black p-2 rounded-md text-gray-400"
                data-testid="button-open"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open menu
                </span>
                <svg
                  class="undefined"
                  fill="inherit"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3.17188 5H20.0156C20.1102 5 20.1875 5.07734 20.1875 5.17188V6.375C20.1875 6.46953 20.1102 6.54688 20.0156 6.54688H3.17188C3.07734 6.54688 3 6.46953 3 6.375V5.17188C3 5.07734 3.07734 5 3.17188 5ZM3.17188 11.5469H20.0156C20.1102 11.5469 20.1875 11.6242 20.1875 11.7188V12.9219C20.1875 13.0164 20.1102 13.0938 20.0156 13.0938H3.17188C3.07734 13.0938 3 13.0164 3 12.9219V11.7188C3 11.6242 3.07734 11.5469 3.17188 11.5469ZM20.0156 18.0938H3.17188C3.07734 18.0938 3 18.1711 3 18.2656V19.4688C3 19.5633 3.07734 19.6406 3.17188 19.6406H20.0156C20.1102 19.6406 20.1875 19.5633 20.1875 19.4688V18.2656C20.1875 18.1711 20.1102 18.0938 20.0156 18.0938Z"
                    fill="#FFFFFF"
                    fill-rule="evenodd"
                    stroke="#FFFFFF"
                    stroke-width="0.25"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden xl:flex-1 xl:block xl:self-stretch"
            >
              <div
                class="h-full flex space-x-10"
              >
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="/uber-uns"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Über uns
                  </span>
                </a>
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="/roesterei"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Unsere Röster
                  </span>
                </a>
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="/blog"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Blog
                  </span>
                </a>
              </div>
            </div>
            <a
              class="flex h-full items-center pl-2 pr-2"
              href="/"
            >
              <span
                class="sr-only"
              >
                60beans
              </span>
              <img
                alt="60beans"
                class="h-6"
                height="24px"
                src="60beans-light.svg"
                width="118px"
              />
            </a>
            <div
              class="flex-1 flow-root flex h-full items-center"
            >
              <div
                class="flex-1 flex h-full items-center justify-end xl:flex-none"
              >
                <div
                  class="hidden h-full xl:flex space-x-8 justify-end"
                >
                  <a
                    class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                    href="/taste-finder"
                  >
                    <span
                      class="uppercase text-sm leading-5"
                    >
                      Tastefinder
                    </span>
                  </a>
                </div>
                <a
                  class="hidden xl:block ml-10"
                  href="//staging.www.60beans.com/profile"
                >
                  <svg
                    class="undefined"
                    fill="inherit"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16 8.66663C16 10.8766 14.21 12.6666 12 12.6666C9.79 12.6666 8 10.8766 8 8.66663C8 6.45663 9.79 4.66663 12 4.66663C14.21 4.66663 16 6.45663 16 8.66663ZM4 18.6666C4 16.0066 9.33 14.6666 12 14.6666C14.67 14.6666 20 16.0066 20 18.6666V20.6666H4V18.6666Z"
                      fill="#FFFFFF"
                      fill-rule="evenodd"
                      stroke="none"
                      stroke-width="none"
                    />
                  </svg>
                </a>
                <div
                  class="ml-4 flow-root xl:ml-10"
                >
                  <button
                    class="relative inline-flex flex-row items-center rounded-full border-2 px-3 py-2 text-sm font-semibold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
                    data-testid="button-shop"
                    role="button"
                  >
                    Shop
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <main
      class="flex flex-col"
      style="min-height: calc(100vh - 66px);"
    >
      <div
        class="flex flex-col md:grid md:grid-cols-3 md:grid-rows-1"
        style="height: calc(100vh - 66px);"
      >
        <div
          class="flex items-center justify-center py-24 mt-4 relative md:justify-start md:order-2 md:py-8 md:mt-0"
        >
          <img
            alt="Tastefinder"
            class="absolute top-16 max-h-64 md:top-auto md:max-h-96 md:-left-16 xl:max-h-96 xl:-left-32"
            src="welcome.webp"
          />
        </div>
        <div
          class="grid items-center justify-center pt-10 pb-44 bg-brand-black text-white md:order-1 md:py-8 md:grid-cols-10 md:col-span-2 xl:grid-cols-5"
        >
          <div
            class="flex flex-col col-items-center md:col-start-2 md:col-end-7 md:order-1 xl:col-start-2 xl:col-end-4"
          >
            <h1
              class="flex w-80 mt-32 font-syne md:mt-0 text-3xl leading-10 font-semibold tracking-tight"
            >
              Finde jetzt deinen Specialty Coffee mit dem Tastefinder.
            </h1>
            <div
              class="flex w-80 mt-4"
            >
              <svg
                aria-hidden="true"
                class="w-12 h-6 mr-4"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
                  fill-rule="evenodd"
                />
              </svg>
              <p
                class="text-base leading-6"
              >
                Finde deinen persönlichen Kaffeetyp und erhalte Empfehlungen, die zu deinem Geschmack passen.
              </p>
            </div>
            <div
              class="flex w-80 mt-8 xl:mt-24"
            >
              <button
                class="flex w-full justify-center relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
                data-testid="button-start"
                role="button"
                type="button"
              >
                Tastefinder starten
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
